src/hdtPhysicsProfiler.cpp

```
@@ -0,0 +1,524 @@
+		const auto msPerFrame = a_frames > 0 ? totalMs / static_cast<double>(a_frames) : 0.0;
+		const auto msPerCall = a_calls > 0 ? totalMs / static_cast<double>(a_calls) : 0.0;
+		const auto pct = a_parentNs > 0 ? static_cast<double>(a_ns) * 100.0 / static_cast<double>(a_parentNs) : 0.0;
+
+		if (a_calls > 0) {
+			logger::info("{:<80} {:>12.3f} ms {:>10.3f} ms/frame {:>10.3f} ms/call {:>8.2f}% {:>8}", a_scope, totalMs, msPerFrame, msPerCall, pct, a_calls);
+		} else {
+			logger::info("{:<80} {:>12.3f} ms {:>10.3f} ms/frame {:>17} {:>8.2f}% {:>8}", a_scope, totalMs, msPerFrame, "-", pct, "-");
+		}
+	}
+
+	static void logTreeRecursive(const AggregateNode& a_node, std::string a_prefix, std::uint32_t a_depth, std::uint64_t a_frames, const DumpOptions& a_options)
```

Cppcheck (2.20.0)

[performance] Function parameter 'a_prefix' should be passed by const reference.

(passedByValue)